docs/reference_guides/core/costing/costing_framework.rst

```diff
@@ -0,0 +1,70 @@
+IDAES Process Costing Framework
+===============================
+
+.. contents:: Contents 
+    :depth: 4
+
+Introduction
+------------
+
+.. rubric:: Available Costing Libraries
+
+This document outlines IDAES support for general process costing and integrating custom costing packages into IDAES models. The base costing framework supports attaching externally-defined blocks and methods for flowsheet and unit model cost calculations. Users must provide a costing package to use for cost calculations; IDAES provides a set of preexisting :ref:`Available Costing Libraries<reference_guides/core/costing/costing_libraries:Available Costing Libraries>`.
+
+When an instance of a unit model is passed to a `UnitModelCostingBlock`, a new sub-block is created 
```

Comment:
This is too much detail up front. For one, readers don't even know what a UnitModelCostingBlock is yet.

There needs to be a general explanation that leads users through the workflow, preferably with a step by step example in code. So start by describing how a user picks a costing package and adds it to a flowsheet/model (the costing parameter block). Then describe how to add capital costing for a specific unit in the process; this will need multiple examples for using a automatic lookup, versus providing a specific method from a library, and also providing a custom written method.

This also needs to cover how to add process material and utility costs for units, which should be done in a separate section from the capital costing.

Then, note that we need to aggregate the costs from all the different units, and show how the framework automates that for you, and where the costs end up.